experimental/Origami/src/action.jl

```diff
@@ -0,0 +1,41 @@
+# see https://arxiv.org/pdf/1809.10327.pdf Lemma 5.4
+
+function action_s(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    return origami_disconnected(v^-1, h, degree(o))
+end
+
+function action_t(o::Origami)
+    h = horizontal_perm(o)
+    return origami_disconnected(h, h^-1 * vertical_perm(o), degree(o))
+end
+
+function action_t_inv(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    return origami_disconnected(h, h * v, degree(o))
+end
+
+function action_s_inv(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    return origami_disconnected(v, h^-1, degree(o))
+end
+
+function action_sl2(A::ZZMatrix, o::Origami)
+    # TODO necessary? loses performance
+    sl2z = special_linear_group(2, ZZ)
+    if !(A in sl2z)
+        throw("Matrix has to be an element of SL2(Z)!")
+    end
```

Comment:
No need to create the group just to check the determinant. Also, nowadays I'd use the `@req` macro
```suggestion
    @req is_one(det(A)) "matrix has to be an element of SL2(Z)"
```